internal/service/models/exceptions/userAlreadyExistErr.go

```diff
@@ -1,9 +1,9 @@
 package exceptions
 
 type UserAlreadyExistErr struct {
-	Message string
+	Err error
 }
 
 func (err *UserAlreadyExistErr) Error() string {
-	return err.Message
+	return err.Err.Error()
 }
```

Comment:
I created multiple errors file just to catch them specifically when needed, for example the databaseConnectionError, this way I can handle them in a better way, instead of just use a Generic error for everything